src/client/datascience/interactive-window/nativeInteractiveWindow.ts

```diff
@@ -474,18 +484,20 @@ export class NativeInteractiveWindow implements IInteractiveWindowLoadable {
 
     protected async submitCode(
         code: string,
-        _file: string,
-        _line: number,
-        _id?: string,
-        _data?: nbformat.ICodeCell | nbformat.IRawCell | nbformat.IMarkdownCell,
-        _debugInfo?: { runByLine: boolean; hashFileName?: string },
+        file: string,
+        line: number,
+        id?: string,
+        data?: nbformat.ICodeCell | nbformat.IRawCell | nbformat.IMarkdownCell,
+        debugInfo?: { runByLine: boolean; hashFileName?: string },
         _cancelToken?: CancellationToken
     ): Promise<boolean> {
+        // Ensure we have a controller to execute code against
+        // and a NotebookDocument to add the NotebookCell to
         const notebookDocument = await this.tryGetMatchingNotebookDocument();
+        await this.initialControllerSelected.promise;
```

Comment:
@rebornix This is the fix for the problem where clicking Run Cell sometimes results in the code not being executed.